### ISSUE TITLE
Make stack switching compatible with address sanitizer

### ISSUE THREAD

**Issue by Keno** [JuliaLang/julia#13541]
Right now we set `-mllvm -asan-stack=0` because address sanitizer does not know what to do with our stack switching. It would be good to fix that such that we can take advantage of this capability. My best guess as to how to fix it is that we'd have to also allocate a shadow map for each stack and then also save the shadow map and restore it when we switch back.

**Comment by vtjnash** [comment]
does msan let you do that? there's also https://github.com/JuliaLang/julia/pull/13099 which fixes the !COPY_STACKS build flag and never messes with the stack.

**Comment by Keno** [comment]
Yes, it does. You have to be a little bit careful, but you can ask it for a pointer to it's shadow memory and use an uninstrumented memcpy.
